### PR TITLE
feat: improve events list ux in dashboard

### LINKED ISSUE #2014: Dashboard event order and UX
### Discuss your ideas or share your views: 

By default the events should appear in descending order of date, since the upcoming events are likely the ones a user is interested in.

Also, it might be useful to separate out the cancelled events from the active ones.

### PR DESCRIPTION
**Pull request by gikf**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #2014

---

- Orders events in dashboard in descending order of date.
- Adds switch to not include canceled events (displayed by default).
- Added parameter to the query makes it awkward, that refetching will not refetch both combinations without doing it explicitly :eyes:.
- One inconvenience is that during using the switch for the first time, it will disappear while loading new list from server. I'm not too enthusiastic about dealing with that problem in this PR. 
- I guess, as an alternative, we could do the filtering just in the client.

### CODE REVIEW

File: client/src/modules/dashboard/Events/pages/EventsPage.tsx [hunk 1]
```diff
@@ -24,6 +55,10 @@ export const EventsPage: NextPageWithLayout = () => {
     <VStack data-cy="events-dashboard">
       <Flex w="full" justify="space-between">
```
Comment (Sboonny) [inline client/src/modules/dashboard/Events/pages/EventsPage.tsx:56]:
```suggestion
      <Flex w="full" justify="space-between" alignItems={{ base: "", sm: "center" }} flexDirection={{base:"column", sm:"row"}}>
```

File: client/src/modules/dashboard/Events/pages/EventsPage.tsx [hunk 2]
```diff
@@ -12,8 +21,30 @@ import { useAuth } from '../../../auth/store';
...
+const ShowCanceledSwitch = ({
+  setShowCanceled,
+  defaultChecked,
+}: {
+  setShowCanceled: React.Dispatch<React.SetStateAction<boolean>>;
+  defaultChecked: boolean;
+}) => {
+  return (
+    <Flex>
+      <FormLabel htmlFor="show-canceled-events">Show canceled events</FormLabel>
```
Comment (Sboonny) [inline client/src/modules/dashboard/Events/pages/EventsPage.tsx:33]:
```suggestion
      <FormLabel htmlFor="show-canceled-events" marginBottom="0">Show canceled events</FormLabel>
```

**Review by Sboonny — COMMENTED** [review]
Here are some styles, to change its looks to this, in mobile devices.

image

File: client/src/modules/dashboard/Events/pages/EventsPage.tsx (hunk 2 above)
Comment (gikf) [reply client/src/modules/dashboard/Events/pages/EventsPage.tsx:33]: I'm not sure about this change, the `0` makes it a bit squeezed.
Before:
image
After:
image

Comment (Sboonny) [reply client/src/modules/dashboard/Events/pages/EventsPage.tsx:33]: the squeeze to zero is for the alignment in layout bigger than `base`, we can go with `marginBottom={{base: "", sm: "0"}}`

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Pull request merged**